synchronization/src/main/java/de/cyface/synchronization/HttpConnection.java

```diff
@@ -186,21 +193,33 @@ public HttpResponse post(@NonNull final HttpURLConnection connection, @NonNull f
     @NonNull
     @Override
     public HttpResponse post(@NonNull final HttpURLConnection connection, @NonNull final File transferTempFile,
-            @NonNull final SyncAdapter.MetaData metaData, @NonNull final String fileName,
+            @NonNull final File eventsTransferTempFile, @NonNull final SyncAdapter.MetaData metaData,
+            @NonNull final String fileName, @NonNull final String eventsFileName,
             @NonNull final UploadProgressListener progressListener)
             throws SynchronisationException, BadRequestException, UnauthorizedException, InternalServerErrorException,
             ForbiddenException, EntityNotParsableException, ConflictException, NetworkUnavailableException,
             SynchronizationInterruptedException, TooManyRequestsException {
 
         // Generate header
         // Attention: Parts of the header (Content-Type, boundary, request method, user agent) are already set
-        final String remainingHeader = generateHeader(metaData, fileName);
+        final String remainingHeader = generateHeader(metaData);
         final byte[] remainingHeaderBytes = remainingHeader.getBytes();
 
+        // File parts
```

Comment:
All the following stuff seems a little duplicated. Why do you not just encapsulate this in a function like (addFile)?

Reply:
We have to calculate the number of bytes written to the connection before starting to actually write bytes so we need to generate the fileHeaderParts in advance.

But I extracted the code which actually writes to the outputStream to an own function, see latest commit.